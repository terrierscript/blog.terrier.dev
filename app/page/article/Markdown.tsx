import React, { FC } from "react"
import ReactMarkdown from "react-markdown"
import { Prism as SyntaxHighlighter } from "react-syntax-highlighter"
import style from "react-syntax-highlighter/dist/cjs/styles/prism/tomorrow"
import { nl2brRemark } from "./unified/nl2br"

export const CodeBlock = ({ value, language }) => {
  return (
    <SyntaxHighlighter language={language} style={style}>
      {value}
    </SyntaxHighlighter>
  )
}

export const RawMarkdown: FC<{ markdown: string }> = ({ markdown }) => {
  return (
    <ReactMarkdown
      source={markdown}
      escapeHtml={false} // for web components
      renderers={{ code: CodeBlock }}
      plugins={[nl2brRemark]}
    />
  )
}
