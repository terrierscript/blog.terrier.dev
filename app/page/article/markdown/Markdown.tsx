import React, { FC } from "react"
import ReactMarkdown from "react-markdown/with-html" // for web components
import { nl2brRemark } from "./unified/nl2br"
import styled from "@emotion/styled"
import { PrismCodeBlock } from "./CodeBlock"
import {
  ARTICLE_COLOR,
  // TITLE_COLOR,
  ARTICLE_TITLE_COLOR
  // FEED_LINK_COLOR
} from "../../../layout/global/colors"

export const CodeWrapper = styled.div`
  margin-bottom: 1.8em;
`

const Link = styled.a`
  text-decoration: underline;
`
const ArticleWrapper = styled.div`
  font-size: 0.9em;
  color: ${ARTICLE_COLOR};
  line-height: 1.3em;
  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    margin: 1em 0;
    color: ${ARTICLE_TITLE_COLOR};
  }
`

const CodeBlock = props => (
  <CodeWrapper>
    <PrismCodeBlock {...props} />
  </CodeWrapper>
)

const List = styled.ul`
  margin-bottom: 0.25em;
`
const ListItem = styled.li`
  margin-top: 0.25em;
`
const Paragraph = styled.p`
  margin-bottom: 0.8em;
`

const InlineCode = styled.span`
  border: 1px solid #ccc;
  background: #eee;
  border-radius: 4px;
  padding: 0.1em 0.4em;
  margin: 0.1em;
`

export const RawMarkdown: FC<{ markdown: string }> = ({ markdown }) => {
  return (
    <ArticleWrapper>
      <ReactMarkdown
        source={markdown}
        escapeHtml={false} // for web components
        renderers={{
          paragraph: Paragraph,
          code: CodeBlock,
          link: Link,
          list: List,
          listItem: ListItem,
          inlineCode: InlineCode
        }}
        plugins={[nl2brRemark]}
      />
    </ArticleWrapper>
  )
}
