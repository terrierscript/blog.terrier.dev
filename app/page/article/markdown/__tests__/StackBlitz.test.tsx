import React from "react"
import renderer from "react-test-renderer"
import { RawMarkdown } from "../Markdown"

describe("StackBlitz iframe", () => {
  it("sample", () => {
    const example = "https://stackblitz.com/edit/react-ts-egg6dg"
    const tree = renderer.create(<RawMarkdown markdown={example} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
  it("sample with params", () => {
    const example =
      "https://stackblitz.com/edit/react-ts-egg6dg?embed=1&file=index.tsx&hideExplorer=1&view=preview&ctl=1"
    const tree = renderer.create(<RawMarkdown markdown={example} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
