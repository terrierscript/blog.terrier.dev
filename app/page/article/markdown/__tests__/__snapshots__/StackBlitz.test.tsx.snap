// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StackBlitz iframe sample 1`] = `
.emotion-5 {
  font-size: 0.9em;
  color: #4a4646;
  line-height: 1.3em;
}

.emotion-5 h1,
.emotion-5 h2,
.emotion-5 h3,
.emotion-5 h4,
.emotion-5 h5,
.emotion-5 h6 {
  margin: 1em 0;
  color: #222;
}

.emotion-4 {
  margin-bottom: 0.8em;
  line-height: 1.7em;
}

.emotion-3 {
  width: 100%;
}

.emotion-1 {
  display: block;
  color: #918b89;
  background: #d6d2d0;
  padding: 0.1em 1em;
  font-size: 0.8em;
  width: 100%;
}

.emotion-0 {
  color: #918b89;
}

.emotion-2 {
  width: 100%;
  min-height: 300px;
  border: 1px solid #d6d2d0;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-4"
  >
    <div
      className="emotion-3"
    >
      <div
        className="emotion-1"
      >
        <a
          className="emotion-0"
          href="https://stackblitz.com/edit/react-ts-egg6dg"
        >
          https:stackblitz.com/edit/react-ts-egg6dg
        </a>
      </div>
      <iframe
        className="emotion-2"
        src="https://stackblitz.com/edit/react-ts-egg6dg"
      />
    </div>
  </div>
</div>
`;

exports[`StackBlitz iframe sample with params 1`] = `
.emotion-5 {
  font-size: 0.9em;
  color: #4a4646;
  line-height: 1.3em;
}

.emotion-5 h1,
.emotion-5 h2,
.emotion-5 h3,
.emotion-5 h4,
.emotion-5 h5,
.emotion-5 h6 {
  margin: 1em 0;
  color: #222;
}

.emotion-4 {
  margin-bottom: 0.8em;
  line-height: 1.7em;
}

.emotion-3 {
  width: 100%;
}

.emotion-1 {
  display: block;
  color: #918b89;
  background: #d6d2d0;
  padding: 0.1em 1em;
  font-size: 0.8em;
  width: 100%;
}

.emotion-0 {
  color: #918b89;
}

.emotion-2 {
  width: 100%;
  min-height: 300px;
  border: 1px solid #d6d2d0;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-4"
  >
    <div
      className="emotion-3"
    >
      <div
        className="emotion-1"
      >
        <a
          className="emotion-0"
          href="https://stackblitz.com/edit/react-ts-egg6dg?embed=1&file=index.tsx&hideExplorer=1&view=preview&ctl=1"
        >
          https:stackblitz.com/edit/react-ts-egg6dg
        </a>
      </div>
      <iframe
        className="emotion-2"
        src="https://stackblitz.com/edit/react-ts-egg6dg?embed=1&file=index.tsx&hideExplorer=1&view=preview&ctl=1"
      />
    </div>
  </div>
</div>
`;
